store-service/src/main/java/practice/store/product/ProductService.java

```
@@ -61,62 +51,88 @@ public ProductPayload getById(long id) {
         return productsList;
     }
 
-    public void save(ProductPayload productPayload) {
-        checkIfProductIsNotWithdrawFromSale(productPayload.getAvailability(), productPayload.getName(), productPayload.getProductUUID());
-        checkIfProductUuidExist(productPayload.getProductUUID());
-
-        if (productPayload.isHasDiscount()) {
-            checkIfDiscountPercentageIsNotToHigh(productPayload.getDiscountPercentage());
-            checkIfDiscountPercentageIsNotToLow(productPayload.getDiscountPercentage());
-            checkIfFinalPriceIsCorrect(productPayload.getBasePrice(), productPayload.getDiscountPercentage(), productPayload.getFinalPrice());
-            checkIfPriceReductionIsCorrect(productPayload.getBasePrice(), productPayload.getDiscountPercentage(), productPayload.getAmountPriceReduction());
-        } else {
-            checkIfDiscountPercentageIsEqualZero(productPayload.getDiscountPercentage());
-            checkIfPriceReductionIsEqualZero(productPayload.getAmountPriceReduction());
-            checkIfFinalPriceAndBasePriceAreEquals(productPayload.getFinalPrice(), productPayload.getBasePrice());
-        }
+    public boolean save(ProductPayload productPayload) {
+        boolean isProductNotWithdrawFromSale = isProductNotWithdrawFromSale(productPayload.getAvailability());
+        boolean isProductUuidExist = isProductUuidExist(productPayload.getProductUUID());
```

right now if the product for the selected uuid does not exist you still perform other methods (line 55,57). That can impact performance in the future. Fail fast would prevent this

---

ok